tests/fuzzer/fuzzer_v2.py

```diff
@@ -0,0 +1,624 @@
+
+
+@dataclass
+class _RollingStock:
+    name: str
+    id: int
+
+
+def run(
+    editoast_url: str,
+    scenario: Scenario,
+    scenario_ttl: int = 20,
+    n_test: int = 1000,
+    log_folder: Optional[Path] = None,
+    infra_name: Optional[str] = None,
+    seed: Optional[int] = None,
+    rolling_stock_name: Optional[str] = None,
+):
+    """
+    Runs every test
+    :param editoast_url: url to reach the editoast api
+    :param scenario: scenario to use for the tests
+    :param scenario_ttl: number of tests to run before a scenario reset
+    :param n_test: number of tests to run
+    :param log_folder: (optional) path to a folder to log errors in
+    :param infra_name: name of the infra, for better reporting
+    :param seed: first seed, incremented by 1 for each individual test
+    :param rolling_stock_name: rolling stock to use, random if None
+    """
+    print("loading infra")
+    infra_graph = _make_graph(editoast_url, scenario.infra)
+    requests.post(editoast_url + f"infra/{scenario.infra}/load").raise_for_status()
+    # The prelude allows us to keep track of path/schedule requests sent so far,
+    # so we can easily reproduce the current state.
+    prelude = []
+    seed = seed or random.randint(0, 2**32)
+    for i in range(n_test):
+        seed += 1
+        print("seed:", seed)
+        random.seed(seed)
+        time.sleep(0.1)
+
+        try:
+            _run_test(infra_graph, editoast_url, scenario, infra_name, prelude, rolling_stock_name)
+        except Exception as e:
+            if log_folder is None:
+                raise e
+            else:
+                print(e)
+                log_folder.mkdir(exist_ok=True)
+                with open(str(log_folder / f"{i}.json"), "w") as f:
+                    print(json.dumps(e.args[0], indent=4, default=lambda o: "<not serializable>"), file=f)
+
+        # Let's reset the scenario (empty timetable) so we can keep a
+        # manageable/reproducible state.
+        if seed % scenario_ttl == 0:
+            scenario = _reset_scenario(editoast_url, scenario)
+            prelude = []
+
+
+def get_infra(editoast_url: str, infra_name: str) -> int:
+    """
+    Returns the ID corresponding to the infra name, if available.
+    :param editoast_url: Api url
+    :param infra_name: name of the infra
+    :return: ID the infra
+    """
+    # TODO: we may want a generic pages handler, if we keep adding queries
+    page = 1
+    while page is not None:
+        r = requests.get(editoast_url + "infra/", params={"page": page})
+        if r.status_code // 100 != 2:
+            raise RuntimeError(f"Infra error {r.status_code}: {r.content}")
+        rjson = r.json()
+        for infra in rjson["results"]:
+            if infra["name"] == infra_name:
+                return infra["id"]
+        page = rjson.get("next")
+    raise ValueError(f"Unable to find infra {infra_name}")
+
+
+def create_scenario(editoast_url: str, infra_id: int) -> Scenario:
```

Comment:
We don't need to have a Project / Study / Scenario anymore. We can have a timetable alone.